chain/src/store.rs

```diff
@@ -479,18 +482,26 @@ impl<'a> Iterator for DifficultyIter<'a> {
 		// Items returned by this iterator cannot be expected to correctly
 		// calculate their own hash - This iterator is purely for iterating through
 		// difficulty information
-		self.header = if self.header.is_none() {
+		let (cur_header, cur_header_hash) = if self.header.is_none() {
 			if let Some(ref batch) = self.batch {
-				batch.get_block_header_skip_proof(&self.start).ok()
+				(
+					batch.get_block_header_skip_proof(&self.start).ok(),
+					Some(self.start.hash()),
```

Comment:
I may be missing something obvious due to not being familiar with Rust. Looking at types `start` seems to be of type `Hash`. Is calling `hash()` on it different than just `Some(self.start)`?
(same question for the `else if` branch below)

Reply:
Had just noticed that the latest block in the TUI 'difficulty' tab was showing up incorrectly, this would be why (hash-of-hash instead of hash). good catch!